static/app/views/performance/utils/useSpanFieldSupportedTags.tsx

```diff
@@ -88,16 +93,19 @@ export function useSpanFieldCustomTags(options?: {
     );
   }, [data]);
 
-  return tags;
+  // XXX: We need to cast here because unwrapping `data` breaks the type returned by
+  //      useQuery above. The react-query library's UseQueryResult is a union type and
+  //      too complex to recreate here so casting the entire object is more appropriate.
+  return {...rest, data: tags} as UseApiQueryResult<TagCollection, RequestError>;
```

Comment:
Hmm, could we just remove the return type from `useSpanFieldCustomTags` and allow it to be inferred so we don't need this casting? Alternatively, if you want to explicitly type the return, it should be something like this:
```
Omit<UseApiQueryResult<SpanFieldsResponse, RequestError>, 'data'> & {
  data: TagCollection;
}
```